codegen/src/main/java/software/amazon/awssdk/codegen/poet/waiters/BaseWaiterClassSpec.java

```diff
@@ -294,4 +328,143 @@ private OperationModel operationModel(WaiterDefinition waiterDefinition) {
+        String operationResponseType = model.getOperation(waiterDefinition.getOperation()).getReturnType().getReturnType();
+        return ClassName.get(modelPackage, operationResponseType);
+    }
+
+    private CodeBlock acceptor(Acceptor acceptor) {
+        CodeBlock.Builder result = CodeBlock.builder();
+
+        switch (acceptor.getState()) {
+            case "success":
+                result.add("$T.success", WaiterAcceptor.class);
+                break;
+            case "failure":
+                result.add("$T.error", WaiterAcceptor.class);
+                break;
+            case "retry":
+                result.add("$T.retry", WaiterAcceptor.class);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported acceptor state: " + acceptor.getState());
+        }
+
+        switch (acceptor.getMatcher()) {
+            case "path":
+                result.add("OnResponseAcceptor(")
+                      .add(pathAcceptorBody(acceptor))
+                      .add(")");
+                break;
+            case "pathAll":
+                result.add("OnResponseAcceptor(")
+                      .add(pathAllAcceptorBody(acceptor))
+                      .add(")");
+                break;
+            case "pathAny":
+                result.add("OnResponseAcceptor(")
+                      .add(pathAnyAcceptorBody(acceptor))
+                      .add(")");
+                break;
+            case "status":
+                // Note: Ignores the result we've built so far because this uses a special acceptor implementation.
+                int expected = Integer.parseInt(acceptor.getExpected().asText());
+                return CodeBlock.of("new $T($L, $T.$L)", WaitersRuntime.ResponseStatusAcceptor.class, expected,
+                                    WaiterState.class, waiterState(acceptor));
+            case "error":
+                result.add("OnExceptionAcceptor(")
+                      .add(errorAcceptorBody(acceptor))
+                      .add(")");
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported acceptor matcher: " + acceptor.getMatcher());
+        }
+
+        return result.build();
+    }
+
+    private String waiterState(Acceptor acceptor) {
+        switch (acceptor.getState()) {
+            case "success":
+                return WaiterState.SUCCESS.name();
+            case "failure":
+                return WaiterState.FAILURE.name();
+            case "retry":
+                return WaiterState.RETRY.name();
+            default:
+                throw new IllegalArgumentException("Unsupported acceptor state: " + acceptor.getState());
+        }
+    }
+
+    private CodeBlock pathAcceptorBody(Acceptor acceptor) {
+        String expected = acceptor.getExpected().asText();
+        String expectedType = acceptor.getExpected() instanceof JrsString ? "$S" : "$L";
```

Comment:
minor: extract to helper method?

Reply:
I feel like customizing the type used in poet is unusual, so putting it away in a helper method would make the reader always have to go to that method to see what it's doing. It seems easier to understand inlined with the method, even if it's less dry.